### ISSUE TITLE
container stuck in stopping state after reboot

### ISSUE THREAD

**Issue by Luap99** [containers/podman#11274]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

After a reboot one of my containers was stuck in the stopping state. I could not start or stop this container anymore. I think systemd killed the podman process on shutdown before the container fully stopped so it got stuck in this state.

**Steps to reproduce the issue:**
Kill the podman stop process while a container is stopping

**Describe the results you received:**
```
$ podman start test
Error: unable to start container "defa11375e41abbe8310aa6921cc53ef9ef7464003355e0b45c09674113f83d6": container defa11375e41abbe8310aa6921cc53ef9ef7464003355e0b45c09674113f83d6 must be in Created or Stopped state to be started: container state improper
$ podman stop test
Error: can only stop created or running containers. defa11375e41abbe8310aa6921cc53ef9ef7464003355e0b45c09674113f83d6 is in state stopping: container state improper
```

**Describe the results you expected:**
The container should be able to start assuming the container process is dead. In my case after a reboot podman should know this and reset the container state to exited.

**Output of `podman version`:**

```
3.2.3
```

**Comment by vrothberg** [comment]
Discussed with @Luap99 on IRC.  Looks like we need to have a look if conmon is still running when doing the "stopping" checks. Cc @mheon

**Comment by mheon** [comment]
This sounds more like a refresh issue to me - the state of the container should be automatically reset at reboot, but it was not?

**Comment by Luap99** [comment]
I am not 100% what was causing this but I rebooted my sever and the service didn't came back so I looked at saw this error. The container is controlled via systemd unit which was created with `podman generate systemd`.

**Comment by mheon** [comment]
Is it still in Stopping state after the reboot?

**Comment by Luap99** [comment]
I recreated the container to get it back working. I cannot guarantee that it was stopping after the reboot, it could also be very well that the container crashed after boot and got into the stopping state.
I tried to reproduce this by killing the podman stop process and I got a container into this stopping state however after I removed the alive file the next podman command did reset the state to `Created` so I guess it works.
Anyway I like @vrothberg suggestion to check if conmon is alive if we run into the stopped state error on podman start at least. I also think podman stop should not error on stopping.

**Comment by mheon** [comment]
I think `podman stop` handles this already - the concerning bit is that we can't even call `podman stop` on a container in Stopping state, which does not seem correct.

**Comment by rhatdan** [comment]
@Luap99 @mheon What is going on with this?

**Comment by Folling** [comment]
Experiencing the same problem on void linux on a virtual machine. Rebooting without manual shutdown leaves containers in `Stopping` state and forces me to manually remove them (and their networks) before starting them up again. 

It also shows "invalid internal status, try resetting the pause process with "podman system migrate": could not find any running process: no such process"

**Comment by rhatdan** [comment]
Open a new issue please with the `podman info` If you are not running with podman 4.4, could you try it out.
